src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationService.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationClient;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationRequest;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationResponse;
+import uk.gov.hmcts.reform.notificationservice.data.Notification;
+import uk.gov.hmcts.reform.notificationservice.data.NotificationRepository;
+
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Service
+public class NotificationService {
+
+    private static final Logger log = getLogger(NotificationService.class);
+
+    private final NotificationRepository notificationRepository;
+    private final ErrorNotificationClient notificationClient;
+
+    public NotificationService(
+        NotificationRepository notificationRepository,
+        ErrorNotificationClient notificationClient
+    ) {
+        this.notificationRepository = notificationRepository;
+        this.notificationClient = notificationClient;
+    }
+
+    public void processPendingNotifications() {
+        List<Notification> notifications = notificationRepository.findPending();
```

Comment:
if there will be many pending records like thousands?,  we may consider to put limit?

Reply:
fair point. i'll consider it as a separate "patch" pr as it is repository layer and not this - service

Reply:
to be honest we are not expecting those high notifications.

Reply:
true, won't hurt to implement. just checked api, not object oriented on this part giving our implementation. so will have to just be simple LIMIT in sql